src/devices/visonic_powercode.c

```diff
@@ -0,0 +1,124 @@
+ * https://metacpan.org/release/Device-RFXCOM/source/lib/Device/RFXCOM/Decoder/Visonic.pm
+ * https://forum.arduino.cc/index.php?topic=289554.0
+*/
+
+#include "decoder.h"
+
+static int visonic_powercode_decode(r_device *decoder, bitbuffer_t *bitbuffer)
+{
+    uint8_t *b; // bits of a row
+    uint8_t msg[32];
+    uint8_t sum[4];
+    data_t *data;
+    char id[7];
+
+    // 37 bits expected, 6 minimum packet repetitions 
+    int row = bitbuffer_find_repeated_row(bitbuffer, 6, 37);
+
+    // exit if incorrect number of bits in row
+    if (bitbuffer->bits_per_row[row] !=37)
+        return DECODE_ABORT_LENGTH;
+
+    // extract message, drop leading start bit and trailing LRC nibble
+    bitbuffer_extract_bytes(bitbuffer, 0, 1, msg, 32);
+
+    // extract LRC nibble
+    bitbuffer_extract_bytes(bitbuffer, 0, 32, sum, 4);
+
+    // TODO: validate checksum here using LRC nibble in sum variable
+
+    // debug
+    if (decoder->verbose > 1) {
+        fprintf(stderr, "%s: data byte is %02x\n",__func__,msg[3]);
+        fprintf(stderr, "%s: check byte is %02x\n",__func__,sum[0]);
+    }
+
+    // format device id
+    sprintf(id, "%02x%02x%02x", msg[0], msg[1], msg[2]); 
+
+    // populate data byte fields
+    /* clang-format off */
+    data = data_make(
+            "model", "Model"          , DATA_STRING, "Visonic-Powercode",
+            "id", "ID"                , DATA_STRING, id,
+            "tamper", "Tamper"        , DATA_INT, ((0x80 & msg[3]) == 0x80) ? 1 : 0,
+            "alarm", "Alarm"          , DATA_INT, ((0x40 & msg[3]) == 0x40) ? 1 : 0,
+            "battery", "Low Battery"  , DATA_INT, ((0x20 & msg[3]) == 0x20) ? 1 : 0,
+            "else", "Else"            , DATA_INT, ((0x10 & msg[3]) == 0x10) ? 1 : 0,
+            "restore", "Restore"      , DATA_INT, ((0x08 & msg[3]) == 0x08) ? 1 : 0,
+            "supervised", "Supervised", DATA_INT, ((0x04 & msg[3]) == 0x04) ? 1 : 0,
+            "spidernet", "Spidernet"  , DATA_INT, ((0x02 & msg[3]) == 0x02) ? 1 : 0,
+            "repeater", "Repeater"    , DATA_INT, ((0x01 & msg[3]) == 0x01) ? 1 : 0,
+            NULL);
+    /* clang-format on */
+
+    // return data
+    decoder_output_data(decoder, data);
+    return 1;
+}
+
+static char *output_fields[] = {
+        "model",
+        "id",
+        "tamper",
+        "alarm",
+        "battery",
+        "else",
+        "restore",
+        "supervised",
+        "spidernet",
+        "repeater",
+        NULL,
+};
+
+r_device visonic_powercode = {
+        .name        = "Visonic powercode",
+        .modulation  = OOK_PULSE_PWM,
+        .short_width = 400,  // short gap is 132 us
+        .long_width  = 800,  // long gap is 224 us
+        .gap_limit   = 900,  // some distance above long
+        .reset_limit = 5000, // a bit longer than packet gap
+        .decode_fn   = &visonic_powercode_decode,
+        .disabled    = 0, // disabled and hidden, use 0 if there is a MIC, 1 otherwise
```

Comment:
disable by default until MIC is implemented

Reply:
fixed